parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -229,20 +230,46 @@ installAzureCNI() {
 }
 
 extractKubeBinaries() {
-    K8S_VERSION=$1
-    KUBE_BINARY_URL=$2
+    local k8s_version="$1"
+    local kube_binary_url="$2"
 
     mkdir -p ${K8S_DOWNLOADS_DIR}
-    K8S_TGZ_TMP=${KUBE_BINARY_URL##*/}
-    retrycmd_get_tarball 120 5 "$K8S_DOWNLOADS_DIR/${K8S_TGZ_TMP}" ${KUBE_BINARY_URL} || exit $ERR_K8S_DOWNLOAD_TIMEOUT
-    tar --transform="s|.*|&-${K8S_VERSION}|" --show-transformed-names -xzvf "$K8S_DOWNLOADS_DIR/${K8S_TGZ_TMP}" \
+    local k8s_tgz_tmp=${kube_binary_url##*/}
+    retrycmd_get_tarball 120 5 "$K8S_DOWNLOADS_DIR/${k8s_tgz_tmp}" ${kube_binary_url} || exit $ERR_K8S_DOWNLOAD_TIMEOUT
+    tar --transform="s|.*|&-${k8s_version}|" --show-transformed-names -xzvf "$K8S_DOWNLOADS_DIR/${k8s_tgz_tmp}" \
         --strip-components=3 -C /usr/local/bin kubernetes/node/bin/kubelet kubernetes/node/bin/kubectl
-    rm -f "$K8S_DOWNLOADS_DIR/${K8S_TGZ_TMP}"
+    rm -f "$K8S_DOWNLOADS_DIR/${k8s_tgz_tmp}"
 }
 
-installKubeletKubectlAndKubeProxy() {
+extractPrivateKubeBinaries() {
```

Comment:
do we need a separate func for this? a lot of the code is same as `extractKubeBinaries`. 

since currently, PRIVATE_KUBE_BINARY_URL is an input from cse_cmd.sh, we can filter the download criteria based on whether that variable is set, right?

Reply:
It can be written multiple ways, but i think the separate function is better - it clearly explains how custom, private and default paths are handled.

Reply:
@anujmaheshwari1 
This is how the common function will look, two separate functions are better, easy to understand what is done in which case and less error prone - but pls confirm if you prefer this version or suggest what you are thinking here? 

extractKubeBinaries() {
    local k8s_version="$1"
    local kube_binary_url="$2"
    local is_private_url="$3"

    local k8s_tgz_tmp=${kube_binary_url##*/}

    local err=$ERR_K8S_DOWNLOAD_TIMEOUT
    if [[ $is_private_url == true ]]; then
        k8s_tgz_tmp="${K8S_CACHE_DIR}/${k8s_tgz_tmp}"
        if [[ -f "${k8s_tgz_tmp}" ]]; then
            echo "cached package ${k8s_tgz_tmp} is found, will use that"
        else
            echo "cached package ${k8s_tgz_tmp} not found"
            return 1
        fi

        # remove the current kubelet and kubectl binaries before extracting new binaries from the cached package
        rm -rf /usr/local/bin/kubelet-* /usr/local/bin/kubectl-*
        err=$ERR_PRIVATE_K8S_PKG_ERR
    fi

    retrycmd_get_tarball 120 5 "${k8s_tgz_tmp}" ${kube_binary_url} || exit "$err"
    tar --transform="s|.*|&-${k8s_version}|" --show-transformed-names -xzvf "${k8s_tgz_tmp}" \
        --strip-components=3 -C /usr/local/bin kubernetes/node/bin/kubelet kubernetes/node/bin/kubectl

    if [[ $is_private_url == false ]]; then
        rm -f "$K8S_DOWNLOADS_DIR/${k8s_tgz_tmp}"
    fi
}

Reply:
i think this works! thank you for consolidating it!